### PR TITLE
refactor(off_policy): change args passing

### PR DESCRIPTION
**Pull request by Gaiejj**
## Description

refactor(off_policy): change args passing

## Motivation and Context

Why is this change required? What problem does it solve?
If it fixes an open issue, please link to the issue here.
You can use the syntax `close #15213` if this solves the issue #15213

- [ ] I have raised an issue to propose this change ([required](https://github.com/PKU-MARL/omnisafe/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://omnisafe.readthedocs.io/en/latest/developer/contributing.html) guide. (**required**)
- [x] My change requires a change to the documentation.
- [x] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [x] I have updated the documentation accordingly.
- [x] I have reformatted the code using `make format`. (**required**)
- [x] I have checked the code using `make lint`. (**required**)
- [x] I have ensured `make test` pass. (**required**)

### CODE REVIEW

**Review by zmsn-2077 — APPROVED** [review]
After careful review, I agree with these changes.

**Review by XuehaiPan — CHANGES_REQUESTED** [review]
Rebase changes.

File: omnisafe/algorithms/off_policy/ddpg.py [hunk 1]
```diff
@@ -35,13 +35,17 @@ class DDPG:  # pylint: disable=too-many-instance-attributes
...
     References:
+
         Title: Continuous control with deep reinforcement learning
+
         Authors: Timothy P. Lillicrap, Jonathan J. Hunt, Alexander Pritzel, Nicolas Heess, Tom Erez,
-                 Yuval Tassa, David Silver, Daan Wierstra.
+                Yuval Tassa, David Silver, Daan Wierstra.
+
         URL: https://arxiv.org/abs/1509.02971
+
```
Comment (XuehaiPan) [inline omnisafe/algorithms/off_policy/ddpg.py:45]: Use items and remove the empty lines:

```text
    References:

        - Title: Continuous control with deep reinforcement learning
        - Authors: Timothy P. Lillicrap, Jonathan J. Hunt, Alexander Pritzel, Nicolas Heess, Tom Erez,
                   Yuval Tassa, David Silver, Daan Wierstra.
        - URL: https://arxiv.org/abs/1509.02971
```

File: omnisafe/algorithms/off_policy/ddpg.py [hunk 2]
```diff
@@ -13,9 +13,9 @@
 # limitations under the License.
 # ==============================================================================
 """Implementation of the DDPG algorithm."""
-
 import time
```
Comment (XuehaiPan) [inline omnisafe/algorithms/off_policy/ddpg.py:16]: An empty line before imports.

File: omnisafe/algorithms/off_policy/ddpg.py [hunk 3]
```diff
@@ -191,36 +193,35 @@ def check_distributed_parameters(self):
...
         """
-        action = self.actor_critic.actor.predict(data['obs'], deterministic=True)
-        loss_pi = self.actor_critic.critic(data['obs'], action)[0]
+        action = self.actor_critic.actor.predict(obs, deterministic=True)
+        loss_pi = self.actor_critic.critic(obs, action)[0]
         pi_info = {}
         return -loss_pi.mean(), pi_info
 
-    def compute_loss_v(self, data):
+    # pylint: disable=too-many-arguments
```
Comment (XuehaiPan) [inline omnisafe/algorithms/off_policy/ddpg.py:207]: Use `pylint: disable-next`, the same changes for all other comments.

File: omnisafe/algorithms/off_policy/ddpg.py [hunk 4]
```diff
@@ -191,36 +193,35 @@ def check_distributed_parameters(self):
...
         Args:
-            data (dict): data dictionary.
-
-        Returns:
-            torch.Tensor.
+            obs (:class:`torch.Tensor`): :meth:`observation` saved in data.
+            act (:class:`torch.Tensor`): :meth:`action` saved in data.
+            rew (:class:`torch.Tensor`): :meth:`reward` saved in data.
+            obs_next (:class:`torch.Tensor`): :meth:`next observations` saved in data.
+            done (:class:`torch.Tensor`): :meth:`terminated` saved in data.
```
Comment (XuehaiPan) [inline omnisafe/algorithms/off_policy/ddpg.py:223]:
```suggestion
            obs (:class:`torch.Tensor`): ``observation`` saved in data.
            act (:class:`torch.Tensor`): ``action`` saved in data.
            rew (:class:`torch.Tensor`): ``reward`` saved in data.
            obs_next (:class:`torch.Tensor`): ``next observations`` saved in data.
            done (:class:`torch.Tensor`): ``terminated`` saved in data.
```

File: omnisafe/algorithms/off_policy/ddpg.py [hunk 5]
```diff
@@ -313,16 +313,42 @@ def learn(self):
...
 
-    def update(self, data):
+    def update(self, data: dict) -> None:
         """Update.
+        Update step contains three parts:
+
+        #.  Update value net by :func:`update_value_net()`
+        #.  Update cost net by :func:`update_cost_net()`
+        #.  Update policy net by :func:`update_policy_net()`
+        #.  Update target net by :func:`polyak_update_target()`
```
Comment (XuehaiPan) [inline omnisafe/algorithms/off_policy/ddpg.py:323]: 1. A extra line before the description.
2. Use `:meth:` without brackets.

```suggestion

        Update step contains three parts:

        #.  Update value net by :meth:`update_value_net`
        #.  Update cost net by :meth:`update_cost_net`
        #.  Update policy net by :meth:`update_policy_net`
        #.  Update target net by :meth:`polyak_update_target`
```

File: omnisafe/algorithms/off_policy/sac.py [hunk 6]
```diff
@@ -39,23 +41,34 @@ def __init__(self, env_id: str, cfgs=None) -> None:
         self.alpha = cfgs.alpha
         self.alpha_gamma = cfgs.alpha_gamma
 
-    # pylint: disable=too-many-locals
-    def compute_loss_v(self, data):
+    # pylint: disable=too-many-locals, disable=too-many-arguments
```
Comment (XuehaiPan) [inline omnisafe/algorithms/off_policy/sac.py:44]:
```suggestion
    # pylint: disable-next=too-many-locals,too-many-arguments
```

File: omnisafe/algorithms/off_policy/sac.py [hunk 7]
```diff
@@ -39,23 +41,34 @@ def __init__(self, env_id: str, cfgs=None) -> None:
...
+        rew: torch.Tensor,
+        obs_next: torch.Tensor,
+        done: torch.Tensor,
+    ) -> Tuple[torch.Tensor, dict]:
         """Computing value loss.
 
-        Args:
-            data (dict): data from replay buffer.
+        .. admonition:: Note
+            :class: hint
```
Comment (XuehaiPan) [inline omnisafe/algorithms/off_policy/sac.py:56]:
```suggestion
        .. hint::
```

File: omnisafe/algorithms/off_policy/ddpg.py [hunk 8]
```diff
@@ -191,36 +191,35 @@ def check_distributed_parameters(self):
...
 
         Args:
-            data (dict): data dictionary.
-
-        Returns:
-            torch.Tensor.
+            obs (:class:`torch.Tensor`): ``observation`` saved in data.
+            act (:class:`torch.Tensor`): ``action`` saved in data.
+            rew (:class:`torch.Tensor`): ``reward`` saved in data.
+            obs_next (:class:`torch.Tensor`): ``net observation`` saved in data.
```
Comment (XuehaiPan) [inline omnisafe/algorithms/off_policy/ddpg.py:220]:
```suggestion
            obs_next (:class:`torch.Tensor`): ``next observation`` saved in data.
```

File: omnisafe/algorithms/off_policy/ddpg.py [hunk 9]
```diff
@@ -313,16 +311,43 @@ def learn(self):
...
-    def update(self, data):
+    def update(self, data: dict) -> None:
         """Update.
 
+        Update step contains three parts:
+
+        #.  Update value net by :meth:`update_value_net()`
+        #.  Update cost net by :meth:`update_cost_net()`
+        #.  Update policy net by :meth:`update_policy_net()`
+        #.  Update target net by :meth:`polyak_update_target()`
```
Comment (XuehaiPan) [inline omnisafe/algorithms/off_policy/ddpg.py:322]:
```suggestion
        Update step contains three parts:

        #.  Update value net by :meth:`update_value_net`
        #.  Update cost net by :meth:`update_cost_net`
        #.  Update policy net by :meth:`update_policy_net`
        #.  Update target net by :meth:`polyak_update_target`
```

No brackets.

Comment (XuehaiPan) [inline omnisafe/algorithms/off_policy/ddpg.py:322]:
```suggestion
        Update step contains three parts:

        #.  Update value net by :meth:`update_value_net`
        #.  Update cost net by :meth:`update_cost_net`
        #.  Update policy net by :meth:`update_policy_net`
        #.  Update target net by :meth:`polyak_update_target`
```

No brackets.

File: omnisafe/algorithms/off_policy/sddpg.py [hunk 10]
```diff
@@ -29,16 +31,16 @@
...
+    """Lyapunov-based Safe Policy Optimization for Continuous Control.
 
     References:
-        Title: Lyapunov-based Safe Policy Optimization for Continuous Control
-        Authors: Yinlam Chow, Ofir Nachum, Aleksandra Faust, Edgar Duenez-Guzman,
-                 Mohammad Ghavamzadeh.
-        URL: https://arxiv.org/abs/1901.10031
+        - Title: Lyapunov-based Safe Policy Optimization for Continuous Control
+        - Authors: Yinlam Chow, Ofir Nachum, Aleksandra Faust, Edgar Duenez-Guzman,
+                Mohammad Ghavamzadeh.
```
Comment (XuehaiPan) [inline omnisafe/algorithms/off_policy/sddpg.py:39]:
```suggestion
        - Authors: Yinlam Chow, Ofir Nachum, Aleksandra Faust, Edgar Duenez-Guzman,
                    Mohammad Ghavamzadeh.
```

Comment (XuehaiPan) [inline omnisafe/algorithms/off_policy/sddpg.py:39]:
```suggestion
        - Authors: Yinlam Chow, Ofir Nachum, Aleksandra Faust, Edgar Duenez-Guzman,
                   Mohammad Ghavamzadeh.
```

**Pull request closed without merging**